### DISCUSSION TITLE
Can this run on openwrt?

### DISCUSSION THREAD

**Discussion by Endlessbugs** [heiher/hev-socks5-tunnel#138] [Q&A] [answered]
I would like to be able to use this software on openwrt by cross compiling, is this possible?

**Comment by heiher** [comment, marked as answer]
Of course. You can build it using openwrt's sdk, like https://github.com/openwrt/packages/tree/master/net/hev-socks5-server

(May be it's time to submit to openwrt packages.

**Reply by Endlessbugs** [reply]
Thx.I'll try.

**Reply by Endlessbugs** [reply]
尝试编译到openwrt上时，安装后报错socks5 tunnel open (No such file or directory)；推测为lib没有被打包进ipk，也没有被安装。我删除了软件Makefile里install相关的选项，因为在交叉编译时会尝试安装，表现为尝试在我的编译环境Ubuntu 24.04上创建目录，然后因权限不足终止编译。我准备给每一个lib创建单独的Makefile并在软件本体的Makefile里声明所需依赖。但是我注意到你在https://github.com/openwrt/packages/tree/master/net/hev-socks5-server中的Makefile 上没有声明任何依赖，这是否意味着有其他的处理办法？

**Reply by heiher** [reply]
`socks5 tunnel open (No such file or directory)` 应该是运行时报错吧？试试在OpenWrt上安装`kmod-tun`:

```sh
opkg install kmod-tun
```

**Reply by Endlessbugs** [reply]
我试试看

**Reply by Endlessbugs** [reply]
能用了，多谢。
如果你看到了被我删除的报错，请不要在意，已经解决。
因为每次升级系统之后，我笔记本电脑无线网卡连接路由器会有bug，所以我习惯性在设置里将网络删除，然后再次连上之后开启了随机mac模式，导致我设置的静态ip分配失效，而我没有注意到这点，我使用Ubuntu虚拟机编译的x86版本进行测试，当我发现 连接虚拟机自身上的socks5服务器通过虚拟机网络连接主机都能正常工作后，虚拟机无法通过路由器局域网连接主机，然后我才去检查了IP地址...太蠢了

**Comment by lilcheti** [comment]
How do you bypass the xray connection?

**Comment by Ser9ei** [comment]
Is it possible to run multiple instances for the different proxies?

**Reply by heiher** [reply]
This project’s OpenWrt packaging does not yet support multiple services, but you can run multiple instances manually.

**Reply by Ser9ei** [reply]
pr https://github.com/openwrt/packages/pull/30143
